domains/certificates/Query.API/API/DataSyncSyncer/DataSyncService.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Security.Claims;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using API.DataSyncSyncer.Dto;
+using CertificateEvents.Primitives;
+using Marten;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace API.DataSyncSyncer;
+
+public class DataSyncService
+{
+    private readonly HttpClient httpClient;
+    private readonly ILogger<DataSyncService> logger;
+    private Dictionary<string, DateTimeOffset>? periodStartTimeDictionary;
+
+    public DataSyncService(HttpClient httpClient, ILogger<DataSyncService> logger)
+    {
+        this.httpClient = httpClient;
+        this.logger = logger;
+    }
+
+    public void SetState(Dictionary<string, DateTimeOffset> state)
+    {
+        periodStartTimeDictionary = state;
+    }
+
+    public async Task<List<DataSyncDto>> FetchMeasurements(string GSRN, string meteringPointOwner,
+        CancellationToken cancellationToken)
+    {
+        var dateFrom = periodStartTimeDictionary[GSRN].ToUnixTimeSeconds();
+
+        var now = DateTimeOffset.UtcNow;
+        var midnight = new DateTimeOffset(now.Year, now.Month, now.Day, 0, 0, 0, TimeSpan.Zero).ToUnixTimeSeconds();
```

Comment:
Suggestion: To increase readability we could create an extension method on DateTimeOffset that returns midnight, so we can have something like:

```
var midnight = DateTimeOffset.UtcNow.GetMidnight();
```